src/spec-configuration/containerFeaturesOCI.ts

```diff
@@ -0,0 +1,197 @@
+import * as path from 'path';
+import * as tar from 'tar';
+import { request } from '../spec-utils/httpRequest';
+import { Log, LogLevel } from '../spec-utils/log';
+import { mkdirpLocal, writeLocalFile } from '../spec-utils/pfs';
+import { FeatureSet } from './containerFeaturesConfiguration';
+
+export interface OCIFeatureRef {
+    id: string;
+    version: string;
+    featureName: string;
+    owner: string;
+    namespace: string;
+    registry: string;
+}
+
+export interface OCIManifest {
+    schemaVersion: number;
+    mediaType: string;
+    config: {
+        mediaType: string;
+        digest: string;
+        size: number;
+    };
+    layers: [
+        {
+            mediaType: string;
+            digest: string;
+            size: number;
+            annotations: {
+                'org.opencontainers.image.ref.name': string;
+                'org.opencontainers.image.title': string;
+            };
+        }
+    ];
+}
+
+export function getOCIFeatureSet(output: Log, identifier: string, options: boolean | string | Record<string, boolean | string | undefined>, manifest: OCIManifest): FeatureSet {
+
+    const featureRef = getFeatureRef(output, identifier);
+
+    const feat = {
+        id: featureRef.id,
+        name: featureRef.id,
+        included: true,
+        value: options
+    };
+
+    let featureSet: FeatureSet = {
+        sourceInformation: {
+            type: 'oci',
+            manifest: manifest,
+            featureRef: featureRef,
+
+        },
+        features: [feat],
+    };
+
+    return featureSet;
+}
+
+export function getFeatureRef(output: Log, identifier: string): OCIFeatureRef {
+
+    // ex: ghcr.io/codspace/features/ruby:1
+    const splitOnColon = identifier.split(':');
+    const id = splitOnColon[0];
+    const version = splitOnColon[1] ? splitOnColon[1] : '1';
+
+    const splitOnSlash = id.split('/');
+    const featureName = splitOnSlash[splitOnSlash.length - 1];
+    const owner = splitOnSlash[1];
+    const registry = splitOnSlash[0];
+    const namespace = splitOnSlash.slice(1, -1).join('/');
+
+    output.write(`identifier: ${identifier}`);
+    output.write(`id: ${id}`);
+    output.write(`version: ${version}`);
+    output.write(`featureName: ${featureName}`);
+    output.write(`owner: ${owner}`);
+    output.write(`namespace: ${namespace}`);
+    output.write(`registry: ${registry}`);
+
+    return {
+        id,
+        version,
+        featureName,
+        owner,
+        namespace,
+        registry
+    };
+}
+
+// Validate if a manifest exists and is reachable about the declared feature.
+export async function validateOCIFeature(output: Log, env: NodeJS.ProcessEnv, identifier: string): Promise<OCIManifest | undefined> {
+    const featureRef = getFeatureRef(output, identifier);
+
```

Comment:
Is this the spot where we should check media type?  I can imagine people trying to use _some_ other oci artifact as a feature and our tooling getting confused.

Reply:
Yep, just to confirm, which one is the correct media type?

Reply:
It depends which layer you're looking at, but as an example the full manifest is:

So we'd probably want to check `config.mediaType` == `application/vnd.devcontainers`.

```
{
  "digest": "sha256:61b4349617cd5eb5b18d5e42498a842b409b8367acf4ade1815fd176587479fd",
  "mediaType": "application/vnd.oci.image.manifest.v1+json",
  "size": 449,
  "config": {
    "digest": "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
    "mediaType": "application/vnd.devcontainers",
    "size": 0
  },
  "layers": [
    {
      "digest": "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
      "mediaType": "application/vnd.oci.image.layer.v1.tar",
      "size": 0
    },
    {
      "digest": "sha256:c33008d0dc12d0e631734082401bec692da809eae2ac51e24f58c1cac68fc0c9",
      "mediaType": "application/octet-stream",
      "size": 16896
    }
  ]
}
```

Reply:
Added the validation.